controllers/stock_location.py

```diff
@@ -27,11 +29,45 @@ def get_location(self, location_id=None, location_name=None, location_barcode=No
         Location = request.env['stock.location']
         identifier = location_id or location_name or location_barcode
         if not identifier:
-            raise ValidationError(_('You need to provide an id, name or barcode for the location.'))
+            raise ValidationError(
+                _('You need to provide an id, name or barcode for the location.'))
 
         location = Location.get_location(identifier)
 
         if check_blocked:
             location.check_blocked()
 
         return location.get_info(extended=True, load_quants=load_quants)[0]
+
+    @http.route('/api/stock-location-pi-count/',
+                type='json', methods=['POST'], auth='user')
+    def pi_count(self, pi_request):
+        """
+            Process a Perpetual Inventory (PI) count request.
+
+            Raises a ValidationError in case of invalid request or if
+            any of the specified location is unknown.
+
+            @param req is a JSON object with the "pi_count_moves",
+            "inventory_adjustments", "preceding_inventory_adjustments"
+            and "location_id" entries
+        """
+        Location = request.env['stock.location']
+
+        location_id = None
+
+        try:
+            location_id = int(pi_request.get('location_id'))
+        except ValueError:
+            pass
+
+        if location_id is None:
+            raise ValidationError(
+                _('You need to provide a valid id for the location.'))
+
+        location = Location.browse(location_id)
```

Comment:
We can use `Location.get_location(location_id)`, then lines 70 and 71 are not needed.